### PR TITLE
:seedling: TK-5067 Add unit test-cases for preflight tools package

### PR DESCRIPTION
**Pull request by prasad-trilio**
Unit testing for preflight package of tools is implemented.

The wait and exec related funcs and packages unit testing is not present in this PR. It will done in a separate PR.

### CODE REVIEW

File: tools/preflight/helper_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+		It("Should return volume-snapshot-class object when snapshotclass name is provided and is present on cluster", func() {
+			vsCRDsMap := map[string]bool{vsClassCRD: true, vsContentCRD: true, vsCRD: true}
+			installVolumeSnapshotCRD(v1beta1K8sVersion, vsCRDsMap)
+			installVolumeSnapshotClass(internal.V1BetaVersion, testDriver, testSnapshotClass)
+			defer func() {
+				deleteSnapshotClass(testSnapshotClass)
+				deleteAllVolumeSnapshotCRD()
+			}()
+			_, err := clusterHasVolumeSnapshotClass(ctx, testSnapshotClass, contClient, k8sClient)
```
Comment (msurudwad) [inline tools/preflight/helper_test.go:115]: check returned object from func.. you can check for `kind`

File: tools/preflight/helper.go [hunk 2]
```diff
@@ -180,6 +186,9 @@ func GetHelmVersion() (string, error) {
 }
 
 func GetServerPreferredVersionForGroup(grp string, cl *goclient.Clientset) (string, error) {
+	if cl == nil {
+		return "", fmt.Errorf("client object is nil, cannot fetch versions of group - %s", grp)
+	}
```
Comment (msurudwad) [inline tools/preflight/helper.go:191]: add this condition where you have initialized this object... no need to check it in every func.
Remove such checks from all utility funcs and remove all test-cases related to this nil check.

File: tools/preflight/cleanup_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+			It("Should delete pod resource of unstructured type", func() {
+				resourceSuffix, err := CreateResourceNameSuffix()
+				Expect(err).To(BeNil())
+				pod := createTestPod(testPodPrefix+resourceSuffix, cleanupOps.Namespace, nil)
+				err = cleanupOps.cleanResource(ctx, pod, contClient)
+				Expect(err).To(BeNil())
+			})
+
+			It("Should return error when resource of unstructured type does not exist in cluster", func() {
```
Comment (msurudwad) [inline tools/preflight/cleanup_test.go:25]: no need to return error.. just log error and continue. Later update test-case.

File: tools/preflight/cleanup_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+				res := &unstructured.Unstructured{}
+				res.SetName(testPodPrefix + testNameSuffix)
+				res.SetNamespace(installNs)
+				res.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind(internal.PodKind))
+				err := cleanupOps.cleanResource(ctx, res, contClient)
+				Expect(err).ToNot(BeNil())
+			})
+		})
+
+		Context("When resource is of structured type", func() {
```
Comment (msurudwad) [inline tools/preflight/cleanup_test.go:35]: structured and unstructured type both are not needed .. deletion functionality is not affected by type.. you can keep only one.. remove other.

File: tools/preflight/cleanup_test.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/trilioData/tvk-plugins/internal"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var _ = Describe("Cleanup Unit Tests", func() {
```
Comment (msurudwad) [inline tools/preflight/cleanup_test.go:11]: need test-cases for func `CleanupPreflightResources`

File: tools/preflight/preflight_test.go [hunk 6]
```diff
@@ -117,8 +117,94 @@ func preflightTestCases(serverVersion string) {
 }
 
 var _ = Context("Preflight Unit Tests", func() {
```
Comment (msurudwad) [inline tools/preflight/preflight_test.go:119]: need test-cases for all major funcs written for each preflight check.. add tests for missing ones

File: tools/preflight/cleanup_test.go (hunk 5 above)
Comment (prasad-trilio) [reply tools/preflight/cleanup_test.go:13]: Will add test-cases for `CleanupPreflightResources` when DNS and volume snapshot resouces checks for env cluster will be added.

File: tools/preflight/preflight_test.go (hunk 6 above)
Comment (prasad-trilio) [reply tools/preflight/preflight_test.go]: Major tests such DNS resolution, volume snapshot check and cleanup with/withhout uid will added in separate ticket

**Review by msurudwad — APPROVED** [review]

**Pull request merged**